Lib/configparser.py

```diff
@@ -659,9 +654,17 @@ def __init__(self, defaults=None, dict_type=_default_dict,
             else:
                 self._optcre = re.compile(self._OPT_TMPL.format(delim=d),
                                           re.VERBOSE)
-        self._prefixes = types.SimpleNamespace(
-            full=tuple(comment_prefixes or ()),
-            inline=tuple(inline_comment_prefixes or ()),
+        comment_prefixes = tuple(comment_prefixes or ())
+        if inline_comment_prefixes:
+            # prefix at the beginning of the line or following a space
+            inline_comment_cre = re.compile(
+                '|'.join(fr'(^|\s)({re.escape(prefix)})'
+                for prefix in inline_comment_prefixes))
+        else:
+            inline_comment_cre = None
+        self._comments = types.SimpleNamespace(
+            full=comment_prefixes,
+            inline=inline_comment_cre,
```

Comment:
Previously, this code was just normalizing and saving some state in one expression. Now it's performing decision logic in three separate blocks. It's complicating the type, by making it asymmetric (full and inline now store different things) and non-deterministic (inline is now optional). Additionally, the logic for where this regular expression is used is now far away from where it's constructed; previously, the regular expression construction was near where it was used and documented by that method.

Reply:
I'll probably add a comment about using `_comments`, thereby mentioning `_LineParser` and the importance of `inline_comment_cre` for performance.

Reply:
The optionality of inline_comment_cre improves performance by not having to search for each line in an empty regex.